src/Aspire.Hosting/Dashboard/DashboardViewModelService.cs

```diff
@@ -90,6 +91,90 @@ await foreach (var tuple in _kubernetesService.WatchAsync<T>(cancellationToken:
         });
     }
 
+    /// <summary>
+    /// This is workaround for https://github.com/dotnet/aspire/issues/1364 where there is a timing issue where an existing connection to Traefik
+    /// will return a 404 for potentially several minutes if that connection is made before it is finished loading its configuration. The below code
+    /// will hit the proxy endpoint setup in that configuration until we do not see the known 404 response from Traefik.
+    /// </summary>
+    private void RunServiceWatchTask()
+    {
+        List<Task> tasks = new List<Task>();
+        _ = Task.Run(async () =>
+        {
+            try
+            {
+                var checkedServices = new HashSet<string>();
+                var maxRetryDuration = TimeSpan.FromSeconds(5);
+                await foreach (var tuple in _kubernetesService.WatchAsync<Service>(cancellationToken: _cancellationToken))
+                {
+                    Task checkTask = Task.Run(async () =>
+                    {
+                        var service = tuple.Item2;
+                        if (service.UsesHttpProtocol(out var uriScheme)
+                            && !checkedServices.Contains(service.Metadata.Name))
+                        {
+                            // We need to verify if service is "actually" working
+                            if (!string.Equals(service.Status?.State, "Ready", StringComparison.OrdinalIgnoreCase))
+                            {
+                                // If service is not ready, we don't try to connect. We will get future notification that service moved to ready state
+                                return;
+                            }
+
+                            var currentTimestamp = DateTime.UtcNow;
+                            var delay = TimeSpan.FromMilliseconds(40);
+                            var errorContent = "404 page not found\n";
+                            var proxyUrlString = $"{uriScheme}://{service.AllocatedAddress}:{service.AllocatedPort}";
+                            while (true)
+                            {
+                                using HttpClient client = new();
```

Comment:
Are we using a new http client on purpose?

Reply:
We want to ensure a new socket connection each request, since traefik holding connections is part of the error condition